modules/combined/test/tests/gap_heat_transfer_mortar/finite-2d/closed_gap_thermomechanical_mortar_contact.i

```diff
@@ -0,0 +1,307 @@
+    type = ADDirichletBC
+    variable = temperature
+    value = 300
+    boundary = 'moving_block_left'
+  []
+  [temperature_right]
+    type = ADDirichletBC
+    variable = temperature
+    value = 800
+    boundary = 'fixed_block_right'
+  []
+[]
+
+[Contact]
+  [interface]
+    primary = moving_block_right
+    secondary = fixed_block_left
+    model = frictionless
+    formulation = mortar
+    use_dual = true
+    correct_edge_dropping = true
+  []
+[]
+
+[Constraints]
+  [thermal_contact]
+    type = ModularGapConductanceConstraint
+    variable = temperature_interface_lm
+    secondary_variable = temperature
+    primary_boundary = moving_block_right
+    primary_subdomain = interface_primary_subdomain
+    secondary_boundary = fixed_block_left
+    secondary_subdomain = interface_secondary_subdomain
+    gap_flux_models = 'closed'
+    use_displaced_mesh = true
+  []
+[]
+
+[Materials]
+  [steel_elasticity_tensor]
+    type = ADComputeIsotropicElasticityTensor
+    youngs_modulus = 1.93e11 #in Pa, 193 GPa, stainless steel 304
+    poissons_ratio = 0.29
+    block = 'left_block'
+  []
+  [steel_stress]
+    type = ADComputeFiniteStrainElasticStress
+    block = 'left_block'
+  []
+  [steel_thermal_expansion]
+    type = ADComputeThermalExpansionEigenstrain
+    thermal_expansion_coeff = 17.3e-6 # stainless steel 304
+    stress_free_temperature = 300.0
+    temperature = temperature
+    eigenstrain_name = 'steel_thermal_expansion'
+    block = 'left_block'
+  []
+  [steel_thermal_properties]
+    type = ADGenericConstantMaterial
+    prop_names = 'steel_density steel_thermal_conductivity steel_heat_capacity steel_hardness'
+    prop_values = ' 8e3            16.2                     0.5                 129' ## for stainless steel 304
+    block = 'left_block'
+  []
+
+  [aluminum_elasticity_tensor]
+    type = ADComputeIsotropicElasticityTensor
+    youngs_modulus = 6.8e10 #in Pa, 68 GPa, aluminum
+    poissons_ratio = 0.36
+    block = 'right_block'
+  []
+  [aluminum_stress]
+    type = ADComputeFiniteStrainElasticStress
+    block = 'right_block'
+  []
+  [aluminum_thermal_expansion]
+    type = ADComputeThermalExpansionEigenstrain
+    thermal_expansion_coeff = 24.0e-6 # aluminum
+    stress_free_temperature = 300.0
+    temperature = temperature
+    eigenstrain_name = 'aluminum_thermal_expansion'
+    block = 'right_block'
+  []
+  [aluminum_thermal_properties]
+    type = ADGenericConstantMaterial
+    prop_names = 'aluminum_density aluminum_thermal_conductivity aluminum_heat_capacity aluminum_hardness'
+    prop_values = ' 2.7e3            210                           0.9                   15' #for 99% pure Al
+    block = 'right_block'
+  []
+[]
+
+[UserObjects]
+  [closed]
+    type = GapFluxModelPressureDependentConduction
+    primary_conductivity = steel_thermal_conductivity
+    secondary_conductivity = aluminum_thermal_conductivity
+    temperature = temperature
+    normal_pressure = interface_normal_lm
```

Comment:
I see you are testing an actual normal pressure here. Disregard my previous comment (which only referred to your first commit).